src/docs/runtime-existential-containers.md

```diff
@@ -0,0 +1,60 @@
+# Existential Containers
+
+In Swift when a protocol *without associated types* or a composition of protocols is passed as an argument into or returned from a function, the value gets passed in an Existential Container. This is a value type which serves as a box for any type that adopts the protocol or protocols needed. It is a [variable length struct](https://github.com/swiftlang/swift/blob/main/docs/ABI/TypeLayout.rst#existential-container-layout) of the form:
+
+| Index | Contents       | Size in Machine Words |
+|-------|----------------|-----------------------|
+|   0   | Payload        | 3                     |
+|   3   | Metadata       | 1                     |
+|   4   | Witness Tables | variable: 0 or more   |
+
+The start of the box includes 3 machine words which serve as a payload for the existential container. If the payload is 3 machine words or less, it will be copied into the existential container. If the payload is larger than 3 machine words, a copy will be allocated from the heap and the pointer to that copy will live in the payload.
```

Comment:
> If the payload is 3 machine words or less, it will be copied into the existential container.

Is this true for `frozen` types only?

Reply:
no because all information you need on how to manipulate the payload will be in the full type metadata (ie, the value witness table). And unless I'm thinking about it incorrectly, ETs are runtime types not compile time types so they are insulated from changes in the types that get boxed.